server/src/main/java/com/objectcomputing/checkins/security/UserPasswordAuthProvider.java

```diff
@@ -20,11 +23,7 @@
     public Publisher<AuthenticationResponse> authenticate(HttpRequest<?> httpRequest, AuthenticationRequest<?, ?> authReq) {
         String username = authReq.getIdentity().toString();
         String password = authReq.getSecret().toString();
-        if (password.equals(store.getUserPassword(username))) {
-            UserDetails details = new UserDetails(username, store.getUserRole(username));
-            return Flowable.just(details);
-        } else {
-            return Flowable.just(new AuthenticationFailed());
-        }
+        UserDetails details = new UserDetails(username, store.getUserRole(password));
```

Comment:
The way it will work is that, the "roleCred" passed in will set the roles, and the username is expected to be the email. 

I will talk more about this in the wiki.